components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/service/BlackListWhiteListService.java

```diff
@@ -291,4 +288,70 @@ public int getAPIId(String providerName, String apiName, String apiVersion) thro
 	public String[] getAPIInfo(int apiId) throws BusinessException {
 		return dao.getAPIInfo(apiId);
 	}
+
+	/**
+	 * Returns the offset and limit required for pagination of blacklisted APIs
+	 *
+	 * @param apiId apiId API ID of blacklisted numbers to get count of
+	 * @param limit maximum number of records per page
+	 * @return
+	 */
+	private List<PaginationDTO> getBlacklistPages(BlacklistWhitelistDTO dto, int limit) throws Exception {
+		int count = dao.getBlacklistCountByApi(dto);
```

Comment:
You might not need this PaginationDTO. Try out with a for loop incremented by 1000000. It will do the same for you